airbyte-server/src/test/java/io/airbyte/server/handlers/WorkspacesHandlerTest.java

```diff
@@ -461,4 +466,42 @@ void testSetFeedbackDone() throws JsonValidationException, ConfigNotFoundExcepti
     verify(configRepository).setFeedback(workspaceGiveFeedback.getWorkspaceId());
   }
 
+  @Test
+  void testWorkspaceIsWrittenThroughSecretsWriter() throws JsonValidationException, ConfigNotFoundException, IOException {
+    secretsRepositoryWriter = mock(SecretsRepositoryWriter.class);
+    workspacesHandler = new WorkspacesHandler(configRepository, secretsRepositoryWriter, connectionsHandler,
+        destinationHandler, sourceHandler, uuidSupplier);
+
+    final UUID uuid = UUID.randomUUID();
+    when(uuidSupplier.get()).thenReturn(uuid);
+
+    final WorkspaceCreate workspaceCreate = new WorkspaceCreate()
+        .name(NEW_WORKSPACE)
+        .email(TEST_EMAIL)
+        .news(false)
+        .anonymousDataCollection(false)
+        .securityUpdates(false)
+        .notifications(List.of(generateApiNotification()))
+        .defaultGeography(GEOGRAPHY_US);
+
+    final WorkspaceRead actualRead = workspacesHandler.createWorkspace(workspaceCreate);
```

Comment:
to confirm - here the written input doesn't have any auth tokens, however this is fine since all we want to confirm is that the write is happening through the SecretRepositoryWriter since the Writer has test confirming that a present token is masked.

Is that right?

Reply:
Yep exactly!